caffe2/opt/optimize_ideep.cc

```diff
@@ -479,6 +602,11 @@ void OptimizeForIdeep(
   enforceFusionInplaceForIdeep(nn);
 
   setPoolingInferenceMode(nn);
+
+  if (training_mode) {
```

Comment:
We do this for non training mode, right?

Reply:
This patch is for training mode.Inference don't need this patch,because mkldnn will cache the primitive format for inference mode.